pylint/checkers/base.py

```diff
@@ -1872,6 +1894,11 @@ def visit_unaryop(self, node):
             operator, right = operand.ops[0]
             if operator not in self.reverse_op:
                 return
+            for _type in (_node_type(left), _node_type(right)):
+                if not _type:
+                    return
+                if isinstance(_type, astroid.Set):
```

Comment:
This will need an explanation why sets are skipped for those not familiar with the fact that they are not naturally ordered.

Another thing that needs to be handled is the case where we have an instance of set which can't be inferred, such as ``builtins.set``, which occurs when we can't infer the arguments to the set call. 

I would also like to have a list of potential nodes or qualified objects which we should skip, in order to not modify the visit function every time we need to add another entry that needs to be skipped.